giskard/datasets/base/__init__.py

```diff
@@ -507,6 +507,19 @@ def upload(self, client: GiskardClient, project_key: str):
             )
         return dataset_id
 
+    def extract_languages(self):
+        """
+        Extracts all languages present in the dataset 'text' column.
+
+        Returns:
+            list[str]: a list of language codescontaining all languages in the dataset.
+        """
+        languages_set = []
+        for col in self.columns:
+            if self.column_types[col] == "text":
+                languages_set = languages_set + self.column_meta[col, self.column_types[col]]['language'].unique().tolist()
```

Comment:
May want to make this a bit more idiomatic, avoiding the for/if nesting and list appending.

E.g. you can get all langs
```py3
langs_per_feature = [
    dataset.column_meta[col, "text"]["language"].dropna().unique()
    for col, col_type in dataset.column_types.items()
    if col_type == "text"
]

return set().union(*langs_per_feature)
```

P.S. `nan` are not handled in your code.